templates/contest/ranking-table.html

```diff
@@ -54,10 +54,18 @@
 
 {% block before_point_head %}
     {% for problem in problems %}
-        <th class="points header"><a href="{{ url('contest_ranked_submissions', contest.key, problem.problem.code) }}">
-            {{- contest.get_label_for_problem(loop.index0) }}
-            <div class="point-denominator">{{ problem.points }}</div>
-        </a></th>
+        <th class="points header">
+            {# problem is a ContestProblem, so get the real problem #}
+            {% if problem.problem.is_accessible_by(request.user) %}
```

Comment:
I wonder if we should use a `Problem.get_visible_problems` and filter by the contest problems so we do 1 database query instead of O(# of contest problem) queries. Also, we should probably give a similar treatment each participation row since they also have links that lead to 404s.

Reply:
Ugh. Yes, we need to give the same treatment to each row.

Reply:
Unfortunately the code that shows that URL is deeply tangled with, of all things, the ContestFormat code.

Reply:
I think `problems = list(contest.contest_problems.select_related('problem').defer('problem__description').order_by('order'))` the line that queries for the problems already gets the problem object, so we aren't hitting the database.

Reply:
https://github.com/DMOJ/online-judge/blob/master/judge/views/contests.py#L646 here I mean

Reply:
Although we don't get the authors, so we might be in an even worth position.

Reply:
Not sure, comments more than welcome.